contracts/GroGvtBridgeSwapper.sol

```diff
@@ -0,0 +1,107 @@
+        IGroController controller = IGroController(_groController);
+
+        require(_stablecoinIndex < 3, "invalid _stablecoinIndex");
+        address _stablecoin;
+        if (_stablecoinIndex == 0) {
+            _stablecoin = controller.DAI();
+        } else if (_stablecoinIndex == 1) {
+            _stablecoin = controller.USDC();
+        } else if (_stablecoinIndex == 2) {
+            _stablecoin = controller.USDT();
+        }
+        stablecoin = _stablecoin;
+        stablecoinIndex = _stablecoinIndex;
+        depositHandler = controller.depositHandler();
+        withdrawHandler = controller.withdrawHandler();
+        gvt = controller.gvt();
+        buoy = controller.buoy();
+        groReferral = _groReferral;
+    }
+
+    function exchange(uint256 _indexIn, uint256 _indexOut, uint256 _amountIn) external override returns (uint256 amountOut) {
+        require(_indexIn + _indexOut == 1, "invalid indexes");
+
+        if (_indexIn == 0) {
+            transferFromZkSync(stablecoin);
+            amountOut = swapStablecoinForGvt(_amountIn);
+            transferToZkSync(gvt, amountOut);
+            emit Swapped(stablecoin, _amountIn, gvt, amountOut);
+        } else {
+            transferFromZkSync(gvt);
+            amountOut = swapGvtForStablecoin(_amountIn);
+            transferToZkSync(stablecoin, amountOut);
+            emit Swapped(gvt, _amountIn, stablecoin, amountOut);
+        }
+    }
+
+    function swapStablecoinForGvt(uint256 _amountIn) public returns (uint256) {
+        uint256[3] memory inAmounts;
+        inAmounts[stablecoinIndex] = _amountIn;
+        uint256 balanceBefore = IGroToken(gvt).balanceOf(address(this));
+
+        IGroToken(gvt).approve(address(depositHandler), _amountIn);
+        uint256 minLpAmount = getMinAmountOut(IGroBuoy(buoy).stableToLp(inAmounts, true));
+        IGroDepositHandler(depositHandler).depositGvt(inAmounts, minLpAmount, groReferral);
+
+        uint256 balanceAfter = IGroToken(gvt).balanceOf(address(this));
+        return balanceAfter - balanceBefore;
+    }
+
+    function swapGvtForStablecoin(uint256 _amountIn) public returns (uint256) {
+        uint256 balanceBefore = IERC20(stablecoin).balanceOf(address(this));
+
+        uint256 usdAmount = IGroToken(gvt).getShareAssets(_amountIn);
+        uint256 lpAmount = IGroBuoy(buoy).usdToLp(usdAmount);
+        uint256 stableAmount = IGroBuoy(buoy).singleStableFromUsd(usdAmount, int128(uint128(stablecoinIndex)));
+        uint256 minAmount = getMinAmountOut(stableAmount);
+        IGroWithdrawHandler(withdrawHandler).withdrawByStablecoin(false, stablecoinIndex, lpAmount, minAmount);
+
+        uint256 balanceAfter = IERC20(stablecoin).balanceOf(address(this));
+        return balanceAfter - balanceBefore;
+    }
+
+    function tokens(uint256 index) external view returns (address) {
+        if (index == 0) {
+            return stablecoin;
+        } else {
```

Comment:
I find this unclean as index must be 0 or 1 or `exchange` will revert, while here we accept any value > 0 for `gvt`.